src/components/List/List.tsx

```diff
@@ -1,90 +1,157 @@
-  rowSize: ListRowSize;
+  label: string;
   /** Width of the list */
   width?: number;
-  /** Height of the list */
+  /** Height of the list when you use it as virtualized */
   height?: number;
   /** Virtualized list option */
   isVirtualized?: boolean;
-  /** Selected Item */
-  selectedItem?: ListItemType;
-  /** Default option. Renders on top of the list, highlighted - for Filter component */
-  defaultOption?: ListItemType;
-  /** Search Term to be highlighted in list items */
-  searchTerm?: string;
-  /** Option Click handler for SelectOption[] data case */
-  handleOptionClick?: SelectHandlerType;
-  /** Defines if this is searchable list or not **/
-  isSearchable?: boolean;
-} & TestProps &
-  React.InputHTMLAttributes<HTMLUListElement>;
+  /** Callback when an item gets a change */
+  onSelectionChange?: (keys: ListSelection) => unknown;
+  /** Is the actual `key` of the item e.g `<Item key={'item_1'} />` is the `item_1` */
+  disabledKeys?: ListSelected;
+  /** Is the actual `key` of the item e.g `<Item key={'item_1'} />` is the `item_1` */
+  selectedKeys?: ListSelected;
+  // /** Search Term to be highlighted in list items */
+  // searchTerm?: string;
+  // /** Defines if this is searchable list or not **/
+  // isSearchable?: boolean;
+} & Omit<AriaListBoxProps<SelectOption>, 'selectionMode' | 'onSelectionChange' | 'children'> &
+  TestProps &
+  Omit<React.InputHTMLAttributes<HTMLUListElement>, 'onChange'>;
 
-const List = React.forwardRef<HTMLUListElement, ListProps>(
-  (
-    {
-      data,
-      rowSize,
-      width,
-      height,
-      isVirtualized = false,
-      selectedItem,
-      isSearchable,
-      defaultOption,
-      searchTerm,
-      handleOptionClick,
-      dataTestId,
-      ...rest
-    },
-    ref
-  ) => {
-    return (
-      <div css={wrapperStyle({ width, isSearchable })}>
-        {isVirtualized ? (
-          <VirtualizedList
-            items={data}
-            rowSize={rowSize}
-            customWidth={width}
-            customHeight={height}
-            ref={ref}
-            selectedItem={selectedItem}
-            defaultOption={defaultOption}
-            searchTerm={searchTerm}
-            handleOptionClick={handleOptionClick}
-            dataTestId={dataTestId}
-            {...rest}
-          />
-        ) : (
-          <NormalList
-            items={data}
-            rowSize={rowSize}
-            width={width}
-            height={height}
-            ref={ref}
-            selectedItem={selectedItem}
-            isSearchable={isSearchable}
-            defaultOption={defaultOption}
-            searchTerm={searchTerm}
-            handleOptionClick={handleOptionClick}
-            dataTestId={dataTestId}
-            {...rest}
-          />
-        )}
+const List = React.forwardRef<HTMLUListElement, ListProps>((props, ref) => {
+  const {
+    width,
+    height,
+    isVirtualized = false,
+    // isSearchable,
+    // searchTerm,
```

Comment:
are this commented on purpose for future PRs? if not, you could delete them.

Reply:
you are right this needs to be handled on `Select` part. I will leave one TODO here

Reply:
solved at [`3b16bc4` (#655)](https://github.com/Orfium/orfium-ictinus/pull/655/commits/514f83b090085ac1faae7a8296bf95ef11827a4e)